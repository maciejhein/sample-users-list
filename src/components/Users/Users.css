.users {
    width: 15rem;
}

.users .users__header {
    text-align: center;
}

.users .users__input {
    width: 100%;
    font-size: 1.1rem;
}
