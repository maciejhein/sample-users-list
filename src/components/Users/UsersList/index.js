import UsersList from "./UsersList";
import "./UsersList.css";

export default UsersList;
