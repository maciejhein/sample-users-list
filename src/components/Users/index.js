import Users from "./Users";
import "./Users.css";

export default Users;
