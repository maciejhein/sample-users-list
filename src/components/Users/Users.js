import React, { useState } from "react";
import PropTypes from "prop-types";

import UsersList from "./UsersList";

const Users = ({ users }) => {
  const [currentUsers, setCurrentUsers] = useState(users.slice(0, 3));

  const handleOnChange = ({ target }) => {
    const searchedUsers = users
      .filter(({ name }) =>
        name.toLowerCase().includes(target.value.toLowerCase())
      )
      .slice(0, 3);

    setCurrentUsers(searchedUsers);
  };
  // TODO Replace input to controled component
  return (
    <section className="users">
      <h1 className="users__header">User List</h1>
      <input
        data-testid="users-input"
        className="users__input"
        type="text"
        placeholder="Search by user name..."
        onChange={handleOnChange}
      />
      <UsersList users={currentUsers} />
    </section>
  );
};

Users.propTypes = {
  users: PropTypes.array.isRequired,
};

export default Users;
