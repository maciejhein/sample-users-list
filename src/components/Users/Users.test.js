import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Users from "./Users";

const users = [
  {
    id: 1,
    name: "John Robinson",
  },
  {
    id: 2,
    name: "John Bradley",
  },
];

describe("COMPONENT - Users", () => {
  it("renders Users component", () => {
    const { asFragment } = render(<Users users={users} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("list with searched users", () => {
    const { getByTestId, container } = render(<Users users={users} />);

    const input = getByTestId("users-input");

    fireEvent.change(input, { target: { value: "Bradley" } });

    const currentUsers = container.querySelectorAll("li");

    expect(currentUsers).toHaveLength(1);
  });
});
