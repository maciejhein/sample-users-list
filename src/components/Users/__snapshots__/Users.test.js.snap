// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Users renders Users component 1`] = `
<DocumentFragment>
  <section
    class="users"
  >
    <h1
      class="users__header"
    >
      User List
    </h1>
    <input
      class="users__input"
      data-testid="users-input"
      placeholder="Search by user name..."
      type="text"
    />
    <ul
      class="users-list"
    >
      <li
        class="users-list__item"
      >
        <span
          class="users-list__item-part"
        >
          1.
        </span>
        <span>
          John Robinson
        </span>
        <span
          class="users-list__item-part"
        >
          @
        </span>
      </li>
      <li
        class="users-list__item"
      >
        <span
          class="users-list__item-part"
        >
          2.
        </span>
        <span>
          John Bradley
        </span>
        <span
          class="users-list__item-part"
        >
          @
        </span>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
